### DISCUSSION TITLE
gzip type file upload

### DISCUSSION THREAD

**Discussion by k0rmo0** [swaggo/swag#1791] [Q&A] [answered]
When i try to upload the gzip file it adds multiform header which breaks my gzip reader because of the Content-Type header

curl -X 'POST' \
  'http://localhost:3013/test' \
  -H 'accept: application/json' \
  -H 'X-REQUEST-SIGNATURE: test' \
  -H 'Content-Type: multipart/form-data' \
  -F 'file=@test.json.gz;type=application/gzip'

I need it to be application/gzip, is there any solution for this problem? Thank you in advance.

**Comment by ubogdan** [comment]
The issue is actually in your go implementation.
a multipart-form data looks like the following. Each element from multipart form has his own content-type as you may actually see int the CURL example command.
```
POST / HTTP/1.1
[[ Less interesting headers ... ]]
Content-Type: multipart/form-data; boundary=---------------------------735323031399963166993862150
Content-Length: 834

-----------------------------735323031399963166993862150
Content-Disposition: form-data; name="text1"

text default
-----------------------------735323031399963166993862150
Content-Disposition: form-data; name="text2"

aωb
-----------------------------735323031399963166993862150
Content-Disposition: form-data; name="file1"; filename="a.txt"
Content-Type: text/plain

Content of a.txt.

-----------------------------735323031399963166993862150
Content-Disposition: form-data; name="file2"; filename="a.html"
Content-Type: text/html

<!DOCTYPE html><title>Content of a.html.</title>

-----------------------------735323031399963166993862150
Content-Disposition: form-data; name="file3"; filename="binary"
Content-Type: application/octet-stream

aωb
-----------------------------735323031399963166993862150--
```

**Reply by k0rmo0** [reply]
I was saying, is there any way I can write in documentation using annotations that I am uploading a gzip file not like form data but like a gzip:

`//	@Param			gzip					formData	file	true	"Application gzip file"`

because I need to upload binary gzip file. On postman this curl works:

`curl --location --request POST 'http://localhost:3013/test' \
--header 'X-REQUEST-SIGNATURE: test==' \
--header 'Authorization: Basic test=' \
--header 'Content-Type: application/gzip' \
--data-binary '@/home/test/test.json.gz'`

**Comment by k0rmo0** [comment, marked as answer]
I had to edit my code and add support for multipart, because i cant send request from swagger without it. 

`	if strings.HasPrefix(req.Header.Get("Content-Type"), "multipart/") {
		r, err = supportMultiParRequest(req)
		if err != nil {
			logger.LogError("Invalid JSON data", err)
			renderer.Renderer.Error(req, res, http.StatusInternalServerError, err.Error())
			return
		}
	} else {
		r, err = gzip.NewReader(req.Body)
		if err != nil {
			logger.LogError("Invalid JSON data", err)
			renderer.Renderer.Error(req, res, http.StatusInternalServerError, err.Error())
			return
		}
	}`

**Comment by ubogdan** [comment]
If you want to use binary, you can use "octet-stream" for	application/octet-stream. On the other hand, application/gzip can be added, but it won't say too much about the content type.
